### ISSUE TITLE
Media query interpolation false positive

### ISSUE THREAD

**Issue by shanpig** [stylelint/stylelint#7816]
### What minimal example or steps are needed to reproduce the bug?

interpolation of ThemeProvider or other imported constants are not supported in the current stylelint. However, it is supported in normal styling.

![image](https://github.com/stylelint/stylelint/assets/31932700/25530321-9d6b-460f-817a-f0a80a6e0a94)

### What minimal configuration is needed to reproduce the bug?

same as top

### How did you run Stylelint?

shift+cmd+p restart stylelint server on vscode

### Which Stylelint-related dependencies are you using?

stylelint ^16.6.1
postcss-styled-syntax ^0.6.4
stylelint-config-standard ^36.0.1

### What did you expect to happen?

interpolation should not be an error.

### What actually happened?

interpolation displays an error.

**Comment by ybiquitous** [comment]
@shanpig Thanks for the report. But, this issue seems to be related to `postcss-styled-syntax` or other CSS-in-JS stuff, not for this Stylelint core repository.

Please consider reporting the issue to them.

**Comment by shanpig** [comment]
Seems like [postcss-styled-syntax doesn't regard this as their issue](https://github.com/hudochenkov/postcss-styled-syntax/issues/32) either 😢

**Comment by ybiquitous** [comment]
@shanpig Okay, can you provide a reproducible repository to help us find this problem?

**Comment by romainmenke** [comment]
https://stylelint.io/demo/#N4Igxg9gJgpiBcIACBbGUCWBDABAChQwDsBaAdwygBcALeHAEmFpjQDoAjAJxiwGsADhGJU2AZxQBfAQA8AlDmAAdIkqqQANhC70A5jxhEA3CskgANCABmGDTAByWNAhAwZTgXbZgxYi+AgiG10XZSIcHCVXGSpDKDEo+gBtFQiIqLEqAE87DRESSCCMXRJMrCIoLC4oKNScAF1Tf0LggDFtFCwqFwArMUD-WAE-RDD0kEycmDyiKkTIkA0umEyo8zqM7Nz8luLSqnLK6vmopdjVkCbJIA

I don't think this is a bug.

This rule specifically detects and warns about invalid/unknown CSS syntax.
`${theme.breakpoint.sm}px` isn't valid CSS.

```css
@media (min-width: ${theme.breakpoint.sm}px) {}
```

You can use ignore comments or disable the rule entirely.

```css
/* stylelint-disable-next-line media-query-no-invalid */
@media (min-width: ${theme.breakpoint.sm}px) {
	color: green;
}
```

**Comment by hudochenkov** [comment]
I think this syntax should be ignored as non-standard with [`isStandardSyntaxAtRule`](https://github.com/stylelint/stylelint/blob/main/lib/utils/isStandardSyntaxAtRule.mjs). Function should be adjusted, though.

**Comment by romainmenke** [comment]
@hudochenkov This rule specifically warns on invalid (and thus non-standard) syntax.
There is no way to correctly and accurately distinguish invalid from non-standard.

The goal of this specific rule is to help users write standard CSS.
If users aren't writing standard CSS then they shouldn't use this rule.

I don't want to make this rule less useful for the most common case: users who write standard CSS.

**Comment by hudochenkov** [comment]
Make sense, re-reading the [rule description](https://stylelint.io/user-guide/rules/media-query-no-invalid/) I agree with @ybiquitous suggestion of disabling the rule or writing a plugin.

> This rule is **only appropriate for CSS**. You **should not** turn it on for CSS-like languages, such as Sass or Less, as they have their own syntaxes.

**Comment by Mouvedia** [comment]
related: #7071

**Comment by Mouvedia** [comment]
I am closing based on https://github.com/stylelint/stylelint/issues/7816#issuecomment-2205936812.
